Holovibes/cuda_includes/masks.cuh

```diff
@@ -0,0 +1,72 @@
+/*! \file
+ *
+ *  \brief Kernels functions to compute masks used in the pipe.
+ *  The differents masks are :
+ *  - Quadratic lens
+ *  - Spectrql lens
+ *  - Cicular mask
+ *  Check the functions to see the different behaviours.
+ */
+#pragma once
+
+#include "common.cuh"
+#include "cuda_memory.cuh"
+
+/*! \brief Compute a lens to apply to an image used by the Fresnel transform (fft1)
+ *
+ * \param output The lens computed by the function.
+ * \param fd File descriptor of the images on which the lens will be applied.
+ * \param lambda Laser dependent wave lenght
+ * \param dist z choosen
+ * \param pixel_size size of pixels of the input
+ */
+__global__ void kernel_quadratic_lens(
+    cuComplex* output, const uint lens_side_size, const float lambda, const float dist, const float pixel_size);
+
+/*! \brief Compute a lens to apply to an image used by the Angular Spectrum (fft2)
+ *
+ * \param output The lens computed by the function.
+ * \param fd File descriptor of the images on wich the lens will be applied.
+ * \param lambda Laser dependent wave lenght
+ * \param distance z choosen
+ * \param pixel_size size of pixels of the input
+ */
+__global__ void kernel_spectral_lens(
+    cuComplex* output, const uint lens_side_size, const float lambda, const float distance, const float pixel_size);
+
+/*! \brief The CUDA Kernel getting a circular mask with a given center and radius.
+ *  Using euclidian distance and circle formula (x^2 +  y^2 = r^2)
+ *
+ *  To call the kernel the threads must be in 2d, follow this example:
+ *  uint threads_2d = get_max_threads_2d();
+ *  dim3 lthreads(threads_2d, threads_2d);
+ *  dim3 lblocks(1 + (width - 1) / threads_2d, 1 + (height - 1) / threads_2d);
```

Comment:
Doxygen comments use markdown so you can put the code in backtips \`\`\`cpp code \`\`\`